pingen_env/models/res_company.py

```diff
@@ -1,46 +1,22 @@
 # Copyright 2023 Camptocamp SA
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
 
-from odoo import api, fields, models
-
-from odoo.addons.server_environment import serv_config
+from odoo import models
 
 
 class ResCompany(models.Model):
-    _inherit = ["res.company"]
-
-    pingen_clientid = fields.Char(
-        compute="_compute_pingen_env", required=False, readonly=True
-    )
-    pingen_client_secretid = fields.Char(
-        compute="_compute_pingen_env", required=False, readonly=True
-    )
-    pingen_organization = fields.Char(
-        compute="_compute_pingen_env", required=False, readonly=True
-    )
-    pingen_staging = fields.Boolean(
-        compute="_compute_pingen_env", required=False, readonly=True
-    )
-    pingen_webhook_secret = fields.Char(
-        compute="_compute_pingen_env", required=False, readonly=True
-    )
+    _name = "res.company"
+    _inherit = ["res.company", "server.env.mixin", "server.env.techname.mixin"]
```

Comment:
FTR using the techname mixin implies that a techname is set on existing records and the env configuration must be updated. As a fallback, I would say that we need a migration script to automatically set the techname = to the current name. Tech name is normalized, so in the script it should be set via SQL if the original name is not normalized.

@grindtildeath am I wrong?

Reply:
You're right on the techname, but there's no way a migration script is enough to have a "transparent" migration, as server environment values will have to be manually updated to reflect the techname anyway.

I'm not sure what's the best way to handle that though...

Reply:
I bet so far it relied on `name`. Just set the techname = name by default and everything should work fine.
Yet, as I mentioned, you should use an sql to update the name and bypass the normalization.
This way, you can avoid to touch the env conf. At least on the 1st upgrade.